internal/configs/configmaps.go

```diff
@@ -399,6 +400,47 @@ func ParseConfigMap(ctx context.Context, cfgm *v1.ConfigMap, nginxPlus bool, has
 		}
 	}
 
+	if zoneSync, exists, err := GetMapKeyAsBool(cfgm.Data, "zone-sync", cfgm); exists {
+		if err != nil {
+			nl.Error(l, err)
+			eventLog.Event(cfgm, v1.EventTypeWarning, nl.EventReasonInvalidValue, err.Error())
+			configOk = false
+		} else {
+			if nginxPlus {
+				cfgParams.ZoneSync.EnableZoneSync = zoneSync
```

Comment:
zone sync is always enabled when using nginx plus even if the passed in config has

```
zone-sync-enabled: false
```
?

Reply:
Or rather, the `zoneSync` variable has the entire `ZoneSync` struct from above, right? In which case does this assignment make sense?

Wouldn't it be better to have

```go
cfgParams.ZoneSync = zoneSync
```
?

Reply:
or perhaps
```go
cfgParams.ZoneSync.EnableZoneSync = zoneSync.EnableZoneSync
```